src/function/gds/rec_joins.cpp

```diff
@@ -202,5 +202,22 @@ void RJAlgorithm::exec(processor::ExecutionContext* executionContext) {
     sharedState->mergeLocalTables();
 }
 
+bool SPAlgorithm::terminateFunc(GDSFrontier& frontier,
+    const processor::NodeOffsetMaskMap& maskMap) {
```

Comment:
is the maskMap here outputMaskMap? If so, let's rename maskMap -> outputMaskMap.

I think we can remove the first if block here.

Reply:
Done